packages/app/src/firebaseApp.ts

```diff
@@ -152,28 +122,15 @@ export class FirebaseAppImpl implements FirebaseApp {
   ): FirebaseService {
     this.checkDestroyed_();
 
-    if (!this.services_[name]) {
-      this.services_[name] = {};
-    }
-
-    if (!this.services_[name][instanceIdentifier]) {
-      /**
-       * If a custom instance has been defined (i.e. not '[DEFAULT]')
-       * then we will pass that instance on, otherwise we pass `null`
-       */
-      const instanceSpecifier =
-        instanceIdentifier !== DEFAULT_ENTRY_NAME
-          ? instanceIdentifier
-          : undefined;
-      const service = this.firebase_.INTERNAL.factories[name](
-        this,
-        this.extendApp.bind(this),
-        instanceSpecifier
-      );
-      this.services_[name][instanceIdentifier] = service;
-    }
-
-    return this.services_[name][instanceIdentifier];
+    // getImmediate will always succeed because _getService is only called for registered components.
+    return (
+      (this.container
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        .getProvider(name as any)
```

Comment:
If you type `name` as `Name`, does it work? I was thinking that the compiler wouldn't complain and that it would understand that it is a `Name` even though it doesn't know all the members in the type `Name`.

Reply:
Yes, but I need to cast it in a different place when calling the function :(

Reply:
Casted it to `Name` and removed the eslint disable line, but it still needs casting because `FirebaseApp` doesn't implement `FirebaseService` interface.  🤷‍♂

Reply:
> `FirebaseApp` doesn't implement `FirebaseService` interface

Haha, nothing we can do about that here.